paperqa/agents/tools.py

```diff
@@ -336,31 +330,53 @@ def extract_answer_from_message(cls, content: str) -> str:
         answer, *rest = re.split(
             pattern=cls.ANSWER_SPLIT_REGEX_PATTERN, string=content, maxsplit=1
         )
-        if len(rest) != 4 or check_could_not_answer(answer):  # noqa: PLR2004
-            return ""
-        return answer
+        return answer if len(rest) == 4 else ""  # noqa: PLR2004
+
+
+class Reset(NamedTool):
+    TOOL_FN_NAME = "reset"
+
+    async def reset(self, state: EnvironmentState) -> None:
+        """
+        Reset by clearing all current evidence from the system.
+
+        This tool is useful when repeatedly failing to answer because new evidence can give new perspective.
```

Comment:
I might say: "This tool is useful when repeatedly failing to answer because the existing evidence may unsuitable for the question."

Reply:
Only because new evidence can always be added, even without using this tool.

Reply:
Nice I see what you're saying, implemented